pkg/aws/aws_client/role.go

```diff
@@ -292,3 +309,58 @@ func completeRolePolicyDocument(statement map[string]interface{}) (string, error
 	assumeRolePolicyDocument, err := json.Marshal(rolePolicyDocument)
 	return string(assumeRolePolicyDocument), err
 }
+
+func (client *AWSClient) AttachPolicy(roleName string, policyArn string, retries int) error {
+	policyAttach := iam.AttachRolePolicyInput{
+		PolicyArn: &policyArn,
+		RoleName:  &roleName,
+	}
+	_, err := client.IamClient.AttachRolePolicy(context.TODO(), &policyAttach)
+	if err != nil {
+		return err
+	}
+
+	start := 0
+	if retries <= 0 {
+		retries = 1
+	}
+
+	for start < retries {
```

Comment:
@OAharoni-RedHat In the cases where retries is `0` or `1`, this adds an unnecessary sleep for 1 minute.

I think the logic should be:

1. Perform the `client.PolicyAttachedToRole` call.
2. If not attached and the user has elected to retry, enter the retry loop
3. Otherwise return the current status via the `error`